### PR TITLE
Add onClick handlers to the schedule and transaction icons in the transaction list (#547)

### PR DESCRIPTION
**Pull request by joel-jeremy**
For #547 

In addition to the arrow icon, I have also updated the recurring/calendar icon to open the edit schedule page.

### CODE REVIEW

**Comment by joel-jeremy** [comment]
Trying to figure out why `useCachedSchedules()` is returning null when run in test (`TransactionTable.test.js`).

**Comment by j-f1** [comment]
> Trying to figure out why `useCachedSchedules()` is returning null when run in test (`TransactionsTable.test.js`).

`useCachedSchedules()` is defined as `useContext(SchedulesContext)`, so it’s likely the thing isn’t being rendered inside a `<SchedulesProvider>` — I think you’ll need to provide that in the test, or handle the schedule array being null, or maybe mock `useCachedSchedules` in tests.

File: packages/desktop-client/src/components/accounts/TransactionsTable.js [hunk 1]
```diff
@@ -480,17 +478,34 @@ function CellWithScheduleIcon({ scheduleId, children }) {
...
+    );
+
+  let onTransferIconClick = () =>
+    !isTemporaryId(transaction.id) &&
+    onNavigateToTransferAccount(transferAccount.id);
+  let TransferDirectionIcon = () =>
+    (transaction._inverse ? -1 : 1) * transaction.amount > 0 ? (
+      <LeftArrow2 style={style} onClick={onTransferIconClick} />
+    ) : (
+      <RightArrow2 style={style} onClick={onTransferIconClick} />
```
Comment (j-f1) [inline packages/desktop-client/src/components/accounts/TransactionsTable.js:500]: Can you make all of these into `<Button bare ...>` to make it clear on hover that you can click? (it might be difficult to get it to fit in layout wise, you may need to use negative margins to make the rows stay the same height as they are now)

File: packages/desktop-client/src/components/accounts/TransactionsTable.js [hunk 2]
```diff
@@ -460,37 +454,64 @@ function PayeeCell({
...
+    ...colorStyle,
   };
 
+  let onScheduleIconClick = () => onNavigateToSchedule(scheduleId);
+
+  let recurring = schedule && schedule._date && !!schedule._date.frequency;
+  let ScheduledIcon = () => (
+    <Button bare style={style} onClick={onScheduleIconClick}>
+      {recurring ? (
+        <ArrowsSynchronize style={colorStyle} />
```
Comment (j-f1) [inline packages/desktop-client/src/components/accounts/TransactionsTable.js:488]: The icons will need to have a `width` property set on their `style` or else they will size themselves to be too big and the buttons will appear blank.

File: packages/desktop-client/src/components/accounts/TransactionsTable.js (hunk 1 above)
Comment (joel-jeremy) [reply packages/desktop-client/src/components/accounts/TransactionsTable.js]: Updated :)

File: packages/desktop-client/src/components/accounts/TransactionsTable.js (hunk 2 above)
Comment (joel-jeremy) [reply packages/desktop-client/src/components/accounts/TransactionsTable.js:488]: Should I just pass the same style used in the button to the icons?

Comment (j-f1) [reply packages/desktop-client/src/components/accounts/TransactionsTable.js:488]: You’d need to pass in a smaller width/height since the button has padding defined on it that reduces the available space for the icon. With the size you’ve got now, the button content area is 13×13px.

Comment (j-f1) [reply packages/desktop-client/src/components/accounts/TransactionsTable.js:488]: Also fwiw the old arrow buttons were 10px wide

Comment (j-f1) [reply packages/desktop-client/src/components/accounts/TransactionsTable.js:488]: …and there wasn’t that 5px padding between the icon and the text (for transfers, schedules did have it. ugh not sure which one to pick here, the schedule icon does look bad™ without the spacing, transfer arrow looks fine but when you hover it is a bit odd)

Comment (joel-jeremy) [reply packages/desktop-client/src/components/accounts/TransactionsTable.js:488]: Adjusted icon sizes to match the existing ones:
image

Hovered:
image

Existing:
image

I think having the same padding for the schedules and transfer resulting in the texts lining up looks better. But let me know if I should adjust it to retain the existing padding.

Comment (j-f1) [reply packages/desktop-client/src/components/accounts/TransactionsTable.js:488]: Having the text line up definitely looks better — good point.

**Comment by MatissJanis** [comment]
(I'll leave this to Jed to approve)

Really like this change! Especially jumping to the transfer account. That's super convenient.

**Comment by youngcw** [comment]
Schedule link is working well for me.

The transfer link makes the account page force scroll to the bottom of the ledger.  If I navigate back to the original account it does the same thing.  This state sticks around and the only way to get it to go away is to close the tab and re-open the page.  This is happening on both chrome and firefox for me

**Review by j-f1 — APPROVED** [review]
Great improvement!

**Comment by joel-jeremy** [comment]
I am encountering the issue @youngcw mentioned. I am thinking maybe it's due to the table navigator scrolling to the focused field? Need to check.

**Comment by j-f1** [comment]
Oop, that comment must have come in between opening the page and approving. I didn’t see anything like that locally, but i only tested with the transfer being the first transaction. Maybe it has something to do with scrolling?

**Comment by joel-jeremy** [comment]
I don't encounter it locally as well but it happens in the netlify demo deployment.
